drivers/ng_at86rf2xx/ng_at86rf2xx_getset.c

```diff
@@ -105,21 +183,52 @@ void ng_at86rf2xx_set_pan(ng_at86rf2xx_t *dev, uint16_t pan)
 
 int16_t ng_at86rf2xx_get_txpower(ng_at86rf2xx_t *dev)
 {
+#ifdef MODULE_NG_AT86RF212B
+    uint8_t txpower = ng_at86rf2xx_reg_read(dev, NG_AT86RF2XX_REG__PHY_TX_PWR);
+    printf("txpower value: %x\n", txpower);
+    return tx_pow_to_dbm(dev->freq, txpower);
+#else
     uint8_t txpower = ng_at86rf2xx_reg_read(dev, NG_AT86RF2XX_REG__PHY_TX_PWR)
                 & NG_AT86RF2XX_PHY_TX_PWR_MASK__TX_PWR;
     return tx_pow_to_dbm[txpower];
+#endif
 }
 
 void ng_at86rf2xx_set_txpower(ng_at86rf2xx_t *dev, int16_t txpower)
 {
+#ifdef MODULE_NG_AT86RF212B
+    txpower += 25;
+#else
     txpower += 17;
+#endif
     if (txpower < 0) {
         txpower = 0;
+#ifdef MODULE_NG_AT86RF212B
+    } else if (txpower > 36) {
+        txpower = 36;
+#elif MODULE_NG_AT86RF233
+    } else if (txpower > 21) {
+        txpower = 21;
+#else
     } else if (txpower > 20) {
         txpower = 20;
-    }
-    ng_at86rf2xx_reg_write(dev, NG_AT86RF2XX_REG__PHY_TX_PWR,
+#endif       
+		} 
```

Comment:
something seems awfully off with `#endif` and the closing brackets here (and indention while we are at it). Pls fix.